### PR TITLE
Feature/ic2 30 implement p rep bonded related functions

### PR DESCRIPTION
**Pull request by dhlee-icon**
- BondedDelegation Calculation logic
- Bond requirement configuration

### CODE REVIEW

File: icon/iiss/icstate/networkvalue.go [hunk 1]
```diff
@@ -118,3 +120,18 @@ func GetTotalStake(s *State) *big.Int {
...
+func (s *State)GetBondRequirement() int {
+	return int(GetBondRequirement(s))
+}
+
+func GetBondRequirement(s *State) int64 {
+	return getValue(s.store, VarBondRequirement).Int64()
+}
+
+func SetBondRequirement(s *State, value int64) error {
+	if value == 0 || value > 100 {
```
Comment (goldworm-icon) [inline icon/iiss/icstate/networkvalue.go:133]: Please consider the condition when value is negative.

Comment (dhlee-icon) [reply icon/iiss/icstate/networkvalue.go]: I reflected the case, thank you.

File: icon/iiss/prepmanager.go [hunk 2]
```diff
@@ -34,28 +34,30 @@ const (
 type PRep struct {
 	*icstate.PRepBase
 	*icstate.PRepStatus
+	*icstate.State
```
Comment (goldworm-icon) [inline icon/iiss/prepmanager.go:37]: Remove icstate.State from PRep and inject bonded requirement from outside.

**Review by goldworm-icon — CHANGES_REQUESTED** [review]

File: icon/iiss/prepmanager.go (hunk 2 above)
Comment (dhlee-icon) [reply icon/iiss/prepmanager.go]: I changed bond requirement value injected to prepStatus through prepManager

File: icon/iiss/icstate/prepstatus.go [hunk 3]
```diff
@@ -116,9 +116,23 @@ func (ps *PRepStatus) SetDelegated(delegated *big.Int) {
 	ps.delegated.Set(delegated)
 }
 
-func (ps *PRepStatus) GetBondedDelegation() *big.Int {
-	// TODO: Not implemented
-	return ps.delegated
+func (ps *PRepStatus) GetBondedDelegation(bondRequirement int) *big.Int {
```
Comment (eunsoo-icon) [inline icon/iiss/icstate/prepstatus.go:119]: Need a comment to help understand

``` go
	// totalDelegation = bond + delegation
	// bondRatio = bond / totalDelegation * 100
	// bondedDelegation = totalDelegation * (bondRatio / bondRequirement)
	//                  = bond * 100 / bondRequirement
	// if bondedDelegation > totalDelegation
	//    bondedDelegation = totalDelegation
```

Change local variable name as per the comments to improve readability

File: icon/iiss/icstate/prepstatus_test.go [hunk 4]
```diff
@@ -16,6 +16,7 @@ package icstate
 import (
 	"github.com/bmizerany/assert"
 	"github.com/icon-project/goloop/common"
```
Comment (eunsoo-icon) [inline icon/iiss/icstate/prepstatus_test.go:18]: move down

**Review by eunsoo-icon — CHANGES_REQUESTED** [review]

File: icon/iiss/icstate/prepstatus.go (hunk 3 above)
Comment (dhlee-icon) [reply icon/iiss/icstate/prepstatus.go:126]: I added the comments and changed name of variables.
Thanks.

File: icon/iiss/icstate/prepstatus_test.go (hunk 4 above)
Comment (dhlee-icon) [reply icon/iiss/icstate/prepstatus_test.go]: I reordered importing package names, thanks

**Review by eunsoo-icon — APPROVED** [review]

**Pull request merged**